<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        function timer(count) {
            return new Promise((resolve, reject) => {
                setTimeout(() => {
                    if(Date.now()%2===1) resolve(`正经时间${count}`);
                    reject(`不正经时间${count}`)
                }, 1000);
            });
        }

         timer(1)
        .then(value=>{console.log('成功的回调',value); return timer(2)})
        .then(value=>{console.log('成功的回调',value); return timer(3)})
        .then(value=>{console.log('成功的回调',value); return timer(4)})
        .then(value=>{console.log('成功的回调',value); return timer(5)})
        .then(value=>{console.log('成功的回调',value); return timer(6)})
        // 当前面链式回调任何一个出现错误都会跳到catch这里
        .catch(reason=>{console.log('发现错误正在执行catch的回调',reason);})

    </script>
</body>
</html>